src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -542,7 +557,8 @@ public int ReadData(Span<byte> buffer)
                 {
                     // Synchronously block waiting for data to be produced.
                     Debug.Assert(bytesRead == 0);
-                    GetWaiterTask().AsTask().GetAwaiter().GetResult();
+                    GetWaiterTask(cancellationToken).GetAwaiter().GetResult();
+                    CancellationHelper.ThrowIfCancellationRequested(cancellationToken);
```

Comment:
Seems like this line should be in GetWaiterTask as the last thing it does?

Reply:
done